### PR TITLE
Add VCH uuid directory during image files browsing

### LINKED ISSUE #8455: Deleting VCH with Shared Image-Store Also Deletes Images for All Other VCHs
#### Summary

When sharing image-store and other VCHs have containers running the VCH cannot be deleted using vic-machine utility. Stopping all containers and shutting down other VCHs the VCH can be deleted. Found doing this deletes the image store.

##### VIC version

1.5.0

#### Details

It should only be that UUID folder that gets deleted if there are other contents in parent folder.

##### Steps to reproduce
1. using vic-machine-linux to create the 1st VCH, using sharedVmfs-0/shared_images as --image-store

./vic-machine-linux create --name VCH_1 --debug 1 --container-name-convention VCH_1_{name} --compute-resource cls --cpu-reservation 1 --cpu-shares normal --memory-reservation 1 --memory-shares normal --endpoint-cpu 1 --endpoint-memory 2048  --image-store sharedVmfs-0/shared_images  --base-image-size 8GB  --volume-store sharedVmfs-0:default --bridge-network bridge --bridge-network-range 172.16.0.0/12 --public-network vm-network --container-network vm-network:cvm-nw --container-network-firewall vm-network:published --tls-cname VCH_1 --certificate-key-size 2048 --no-tlsverify --user Administrator@VSPHERE.LOCAL  --target 10.92.106.103/dc1 --insecure-registry wdc-harbor-ci.eng.vmware.com --password 'Admin!23' --thumbprint 6D:36:77:DC:74:59:5A:45:EC:ED:D2:03:20:91:B6:A1:F4:98:F0:F8

2. using vic-machine-linux to create the 2nd VCH, using sharedVmfs-0/shared_images as --image-store

./vic-machine-linux create --name VCH_2 --debug 1 --container-name-convention VCH_2_{name} --compute-resource cls --cpu-reservation 1 --cpu-shares normal --memory-reservation 1 --memory-shares normal --endpoint-cpu 1 --endpoint-memory 2048  --image-store sharedVmfs-0/shared_images  --base-image-size 8GB  --volume-store sharedVmfs-0:default --bridge-network bridge --bridge-network-range 172.16.0.0/12 --public-network vm-network --container-network vm-network:cvm-nw --container-network-firewall vm-network:published --tls-cname VCH_2 --certificate-key-size 2048 --no-tlsverify --user Administrator@VSPHERE.LOCAL  --target 10.92.106.103/dc1 --insecure-registry wdc-harbor-ci.eng.vmware.com --password 'Admin!23' --thumbprint 6D:36:77:DC:74:59:5A:45:EC:ED:D2:03:20:91:B6:A1:F4:98:F0:F8

3. From ESXi host, we can see two folders are created in sharedVmfs-0/shared_images/VIC, each one is mapping to each VCH

[root@wdc-rdops-vm06-dhcp-117-20:/vmfs/volumes/5c469238-b893c79f-3b91-0200578af940/shared_images/VIC] ls
421e1efb-ac2c-e624-439e-76dc86561b78(VCH1) 421e31d3-3a95-4b62-672b-01959a355564(VCH2)

4. create a busybox container vm in each VCH

4.1 VCH1
docker -H 10.92.106.206:2376 --tls ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
fc8258b2c333        busybox             "sh"                2 minutes ago       Up 8 seconds                            musing_lamarr
4.2 VCH2
docker -H 10.92.124.138:2376 --tls ps
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
f822926c9808        busybox             "sh"                About a minute ago   Up About a minute                       angry_saha

5. Stop container vm in the VCH1 and delete the VCH1

docker -H 10.92.106.206:2376 --tls stop fc8258b2c333
./vic-machine-linux delete --name VCH_1 --user Administrator@VSPHERE.LOCAL  --target 10.92.106.103/dc1 --password 'Admin!23' --thumbprint 6D:36:77:DC:74:59:5A:45:EC:ED:D2:03:20:91:B6:A1:F4:98:F0:F8

##### Actual behavior
```
INFO[0115] Removing appliance VM network devices
INFO[0131] Collecting 4342068d-f079-4bea-997d-4500ef9cf27d vpxd.log
ERRO[0131] Error caused by file /vmfs/volumes/5c469238-b893c79f-3b91-0200578af940/shared_images/VIC/421e31d3-3a95-4b62-672b->01959a355564/images/scratch/scratch.vmdk
```
421e31d3-3a95-4b62-672b-01959a355564 is the one refer to VCH2
when trying to delete VCH1, our code is operating the image folder created by VCH2.
```
ERRO[0132] --------------------
ERRO[0132] vic-machine-linux delete failed: delete failed
```
##### Expected behavior
VCH1 get deleted and nothing is touched on VCH2

### PR DESCRIPTION
**Pull request by wjun**
When multiple VCHs share the same image datastore path, we need to
use VCH uuid directory to distinguish images related to a specific
VCH.

[specific ci=Group6-VIC-Machine.6-04-Create-Basic]
Fix #8455

### CODE REVIEW

File: lib/install/management/store_files.go [hunk 1]
```diff
@@ -83,7 +88,7 @@ func (d *Dispatcher) deleteImages(conf *config.VirtualContainerHostConfigSpec) e
 			continue
 		}
 
-		if len(children) == 0 {
+		if len(children) == 0 || len(children) == 1 { // when image store path is given, we need to delete VIC dir too when it is empty.
```
Comment (yuyangbj) [inline lib/install/management/store_files.go]: Do we still have the scenario len(children)==0?

Comment (wjun) [reply lib/install/management/store_files.go]: Good catch. Remove that condition.

File: lib/install/management/store_files.go [hunk 2]
```diff
@@ -83,7 +88,7 @@ func (d *Dispatcher) deleteImages(conf *config.VirtualContainerHostConfigSpec) e
 			continue
 		}
 
-		if len(children) == 0 {
+		if len(children) == 1 { // when image store path is given, we need to delete VIC dir too when it is empty.
```
Comment (DanielXiao) [inline lib/install/management/store_files.go]: What does children stand for? If it is the number of child folders, is it possible to have any other folders retain here? Or is it thread safe when you delete 2 VCH at the same time.

Comment (wjun) [reply lib/install/management/store_files.go]: Yes, it stands for folders and files under a root directory. The parallel file access is protected by VC itself.

**Pull request merged**